packages/neon/neon/lib/src/widgets/drawer.dart

```diff
@@ -0,0 +1,208 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:meta/meta.dart';
+import 'package:neon/l10n/localizations.dart';
+import 'package:neon/neon.dart';
+import 'package:neon/src/router.dart';
+import 'package:neon/src/widgets/drawer_destination.dart';
+import 'package:provider/provider.dart';
+
+@internal
+class NeonDrawer extends StatelessWidget {
+  const NeonDrawer({
+    super.key,
+  });
+
+  @override
+  Widget build(final BuildContext context) {
+    final accountsBloc = Provider.of<AccountsBloc>(context, listen: false);
+    final appsBloc = accountsBloc.activeAppsBloc;
+
+    return StreamBuilder(
+      stream: appsBloc.appImplementations,
+      builder: (final context, final snapshot) {
+        if (snapshot.data?.data == null) {
+          return Container();
+        }
+
+        return _NeonDrawer(
+          apps: snapshot.data!.data!,
+        );
+      },
+    );
+  }
+}
+
+class _NeonDrawer extends StatefulWidget {
+  const _NeonDrawer({
+    required this.apps,
+  });
+
+  final Iterable<AppImplementation> apps;
+
+  @override
+  State<_NeonDrawer> createState() => __NeonDrawerState();
+}
+
+class __NeonDrawerState extends State<_NeonDrawer> with SingleTickerProviderStateMixin {
+  late TabController _tabController;
+  late AccountsBloc _accountsBloc;
+  late AppsBloc _appsBloc;
+  late List<AppImplementation> _apps;
+
+  int _activeApp = 0;
+
+  @override
+  void initState() {
+    super.initState();
+
+    _accountsBloc = Provider.of<AccountsBloc>(context, listen: false);
+    _appsBloc = _accountsBloc.activeAppsBloc;
+
+    _apps = widget.apps.toList();
+    _activeApp = _apps.indexWhere((final app) => app.id == _appsBloc.activeAppID.valueOrNull);
+
+    _tabController = TabController(
+      vsync: this,
+      length: widget.apps.length,
+    );
+  }
+
+  @override
+  void dispose() {
+    _tabController.dispose();
+    super.dispose();
+  }
+
+  Future<void> onAppChange(final int index) async {
+    if (index < _apps.length) {
+      unawaited(_appsBloc.setActiveApp(_apps[index].id));
+      //context.goNamed(apps[index].routeName);
+
+      setState(() {
+        _activeApp = index;
+      });
+    } else {
+      const SettingsRoute().go(context);
+    }
```

Comment:
This logic isn't that easy to understand. Could you add a comment to explain why this check exists? It took me few moments to understand what is going on.